ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -0,0 +1,777 @@
+   * @param time the Time instance to use.
+   * @throws RouterException if there is an error with any of the parameters, such as an invalid blob id.
+   */
+  GetBlobOperation(RouterConfig routerConfig, ClusterMap clusterMap, ResponseHandler responseHandler, String blobIdStr,
+      FutureResult<ReadableStreamChannel> futureResult, Callback<ReadableStreamChannel> callback,
+      OperationCompleteCallback operationCompleteCallback, Time time)
+      throws RouterException {
+    super(routerConfig, clusterMap, responseHandler, blobIdStr, futureResult, callback, time);
+    this.operationCompleteCallback = operationCompleteCallback;
+    this.blobIdFactory = new BlobIdFactory(clusterMap);
+    firstChunk = new FirstGetChunk(blobId);
+  }
+
+  /**
+   * {@inheritDoc}
+   * <br>
+   * If the callback for the operation was already called, then abort the reads from the channel and the caller
+   * will be notified as part of the read callback eventually. If not, the notification will happen as part of the
+   * operation completion callback.
+   */
+  @Override
+  void abort(Exception abortCause) {
+    if (operationCallbackInvoked.compareAndSet(false, true)) {
+      operationCompleteCallback.completeOperation(operationFuture, operationCallback, null, abortCause);
+    } else {
+      operationException.set(abortCause);
+      if (readCalled) {
+        completeRead();
+      }
+    }
+    operationCompleted = true;
+  }
+
+  /**
+   * Return the {@link MessageFormatFlags} to associate with a getBlob operation.
+   * @return {@link MessageFormatFlags#Blob}
+   */
+  @Override
+  MessageFormatFlags getOperationFlag() {
+    return MessageFormatFlags.Blob;
+  }
+
+  /**
+   * Do all that needs to be done (cleanup, notification, etc.) on chunk completion and mark the state of the chunk
+   * appropriately.
+   * @param chunk the chunk that has completed.
+   */
+  private void onChunkOperationComplete(GetChunk chunk) {
+    if (chunk.getChunkException() != null) {
+      // if operation callback was already called, then this exception will have to be notified as part of the
+      // read callback.
+      setOperationException(chunk.getChunkException());
+    }
+    if (chunk == firstChunk) {
+      if (operationCallbackInvoked.compareAndSet(false, true)) {
+        // Complete the operation from the caller's perspective, so that the caller can start reading from the
+        // channel. The operation will not be marked as complete internally as subsequent chunk retrievals and
+        // channel writes will need to happen and for that, this operation needs the GetManager to poll it
+        // periodically. If any exception is encountered while processing subsequent chunks, those will be notified
+        // during the channel read.
+        ReadableStreamChannel result = getOperationException() == null ? this : null;
+        operationCompleteCallback
```

Comment:
You are right. I don't think there is much we can do about it though. Internally, for metrics, the operation managers could report the count and that will account for all operations. This variable is in the router, and as far as the router is concerned, the operation will be complete. We could see when we add metrics whether this variable has any value. Currently this is just the number of operations that are waiting for a result from the router.